### ISSUE TITLE
Non-actionable feedback for the "analyzer" package

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/pana#515]
Looking at https://pub.dartlang.org/packages/analyzer#-analysis-tab-, the analyzer shows up as having health zero.  The only info in "Analysis issues and suggestions" is as follows:

```
Running dartanalyzer . failed with the following output:

Invalid argument(s): Provided content does not align with expectations.
`STDERR exceeded 100000 lines.`
```

I'm not sure how to act on this, and it leads me to suspect a bug in the pana tool.  When I run the analyzer on itself from the command line, I get a much more sensible result:

```
Analyzing analyzer...
  hint • 'fileContentOverlay' is deprecated and shouldn't be used at lib/src/dart/analysis/analysis_context_collection.dart:47:9 • deprecated_member_use_from_same_package
  hint • 'package:analyzer/source/analysis_options_provider.dart' is deprecated and shouldn't be used at lib/src/lint/analysis.dart:13:1 • deprecated_member_use_from_same_package
  hint • 'file:///usr/local/google/home/paulberry/dart-new/sdk/pkg/analyzer/test/generated/constant_test.dart' is deprecated and shouldn't be used at test/generated/test_all.dart:12:1 • deprecated_member_use_from_same_package
  hint • 'file:///usr/local/google/home/paulberry/dart-new/sdk/pkg/analyzer/test/source/embedder_test.dart' is deprecated and shouldn't be used at test/source/test_all.dart:8:1 • deprecated_member_use_from_same_package
4 hints found.
```

**Comment by stereotype441** [comment]
@isoos @jonasfj FYI

**Comment by isoos** [comment]
@stereotype441: pana overrides the package-provided analysis options file in order to create a uniform metric across packages. [the current](https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/analysis_options.yaml) options file uses very little linters enabled. We are following [stagehand's defaults](https://github.com/dart-lang/stagehand/blob/master/templates/package-simple/analysis_options.yaml) which now uses `pedantic` as the default linter set.

Could you please update the options file and re-run with that?

**Comment by stereotype441** [comment]
@isoos With this as my analysis options file:
```
include: package:pedantic/analysis_options.yaml
```
I still get a reasonable looking result:
```
Analyzing analyzer...
  hint • 'fileContentOverlay' is deprecated and shouldn't be used at lib/src/dart/analysis/analysis_context_collection.dart:47:9 • deprecated_member_use_from_same_package
  hint • 'package:analyzer/source/analysis_options_provider.dart' is deprecated and shouldn't be used at lib/src/lint/analysis.dart:13:1 • deprecated_member_use_from_same_package
  hint • 'file:///usr/local/google/home/paulberry/dart-new/sdk/pkg/analyzer/test/generated/constant_test.dart' is deprecated and shouldn't be used at test/generated/test_all.dart:12:1 • deprecated_member_use_from_same_package
  hint • 'file:///usr/local/google/home/paulberry/dart-new/sdk/pkg/analyzer/test/source/embedder_test.dart' is deprecated and shouldn't be used at test/source/test_all.dart:8:1 • deprecated_member_use_from_same_package
4 hints found.
```

**Comment by isoos** [comment]
You need to add it to the `pubspec.yaml` file as a dependency, and run `pub get`. [See here](https://www.dartlang.org/guides/language/analysis-options#default-google-rules-pedantic)

After doing that and running `dartanalyzer` of `SDK 2.2.0` on the `lib/` directory, I've got the following:

````
[...]
  lint • Prefer using /// for doc comments at lib/src/workspace/workspace.dart:42:3 • slash_for_doc_comments
  lint • Prefer using /// for doc comments at lib/src/workspace/workspace.dart:52:1 • slash_for_doc_comments
  hint • 'fileContentOverlay' is deprecated and shouldn't be used at lib/src/dart/analysis/analysis_context_collection.dart:47:9 • deprecated_member_use_from_same_package
  hint • 'analyzer.src.analysis_options.analysis_options_provider' is deprecated and shouldn't be used at lib/src/lint/analysis.dart:13:1 • deprecated_member_use_from_same_package
7092 lints and 2 hints found.
````

**Comment by isoos** [comment]
This was fixed and released in 0.12.14.
